### PR TITLE
Updated patch using Rest 

### PR DESCRIPTION
**Pull request by ramach**
I believe addressed most from last feedback - currently partitions and tables need more work

Database unit test works

### CODE REVIEW

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java [hunk 1]
```diff
@@ -1,345 +1,318 @@
...
+        if (dbname == null || dbname.isEmpty() || db == null)
+            throw new NoSuchObjectException("Unable to locate database " + dbname + " with " + db);
+        try {
+            dropDatabase(dbname);
+            createDatabase(db);
+        } catch (NoSuchObjectException | MetaException ex) {
+            LOG.debug("Alter database failed with: {}", ex.getMessage());
+            throw ex;
+        } catch (InvalidObjectException ex) {
+            LOG.debug("Alter database failed with: " + ex.getMessage());
```
Comment (akon-dey) [inline hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java]: This should be an error.

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java [hunk 2]
```diff
@@ -1,345 +1,318 @@
...
-      // ignore if the database does not exist
+    @Override
+    public boolean alterDatabase(String dbname, Database db) throws NoSuchObjectException, MetaException {
+        if (dbname == null || dbname.isEmpty() || db == null)
+            throw new NoSuchObjectException("Unable to locate database " + dbname + " with " + db);
+        try {
+            dropDatabase(dbname);
+            createDatabase(db);
+        } catch (NoSuchObjectException | MetaException ex) {
+            LOG.debug("Alter database failed with: {}", ex.getMessage());
```
Comment (akon-dey) [inline hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStore.java]: This should be an error.

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundTable.java [hunk 3]
```diff
@@ -38,312 +31,252 @@
...
-          + nodeVersion.getNodeId());
-    } catch (GroundException e) {
-      LOG.error("Unable to fetch table partition edge structure");
-      throw e;
+    Node getNode(String tableName, Map<String, Tag> tags) throws GroundException {
+        try {
+            logger.debug("Fetching table node: {}", tableName);
+            return groundReadWrite.getGroundReadWriteNodeResource().getNode(tableName);
+        } catch (GroundException ge1) {
+            logger.debug("Not found - Creating table node: " + tableName);
```
Comment (akon-dey) [inline hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundTable.java]: This should be an error log.

**Review by akon-dey — COMMENTED** [review]
I have taken a first pass of the code changes. I have some logging related nits. Please address them. I will continue to go over the change and provide feedback.

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundTable.java (hunk 3 above)
Comment (ramach) [reply hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundTable.java]: Yes. I will change all these

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java [hunk 4]
```diff
@@ -1,33 +1,31 @@
 /**
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (vsreekanti) [inline hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java]: Why is the license formatting changed?

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStoreBase.java [hunk 5]
```diff
@@ -96,37 +69,33 @@ public boolean addPartitions(String dbName, String tblName, List<Partition> part
     return false;
   }
 
-  public boolean addPartitions(String dbName,
-                               String tblName,
-                               PartitionSpecProxy partitionSpec,
-                               boolean ifNotExists)
+  public boolean addPartitions(String dbName, String tblName, PartitionSpecProxy partitionSpec, boolean ifNotExists)
```
Comment (vsreekanti) [inline hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStoreBase.java]: Does this fit within the 100 character line limit?

Comment (vsreekanti) [reply hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStoreBase.java]: True throughout this file.

**Review by vsreekanti — CHANGES_REQUESTED** [review]
Made a first pass through the code. Things look good for the most part. Code is a lot cleaner than before!

Couple high-level nitpicks: 

* There are a bunch of 10-15 line chunks of code without any line breaks. For readability's sake, please break them up.
* There's a bunch of dead whitespace between methods. Again for readability, please clean up. 😄

**Comment by ramach** [comment]
Will fix. thanks again

On Mon, Apr 17, 2017 at 12:44 PM, Vikram Sreekanti <notifications@github.com

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java (hunk 4 above)
Comment (ramach) [reply hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundDatabase.java]: fixed

File: hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStoreBase.java (hunk 5 above)
Comment (ramach) [reply hive-plugin/src/main/java/edu/berkeley/ground/plugins/hive/GroundStoreBase.java]: We never enforced 100 char limit in the old formatting. I have googlestyle with 100 char limit I can use that. But changes will be too many

**Review by vsreekanti — APPROVED** [review]
This looks great, @ramach! Much cleaner and easier to read.

**Pull request merged**